### ISSUE TITLE
Better support for $or operator

### ISSUE THREAD

**Issue by Favorlock** [immutables/immutables#385]
I've been learning how to use immutables mongo today and I've discovered that the way you do the or logic is extremely limiting compared to other libraries. For example, in Morphia the or method accepts an array of criteria and applies it to a field. I would personally like to see support for generating or methods for the various fields. Here is an example of the query generated by Morphia:

`{ uuid: "8b7a881c-6ccb-4ada-8f6a-60cc99e6aa20", action: { $in: [ "BAN", "IPBAN" ] }, $or: [ { expires: { $exists: false } }, { expires: { $gt: new Date(1467364749679) } } ] }`

I can't seem to find any way to make this same query with immutables.

**Comment by elucash** [comment]
out integration, of course, provide only limited set tools for criteria building, but they are quite straightforward, expressive, typesafe, benefitting from auto-completion. We have watched other java query DSLs being quite a mess, not any better than string concatenation. We support JSON strings queries, so you can actually use the example you provided as a query (maybe except for the `new Date` part ;) ) using `Repository#find(String)` method.
Having that said, I've made couple of fixes and refinements (to be released tonight or tomorrow) to the generated criteria, so this query can be expressed as 

```java
EntRepository.Criteria uuidBan = repository.criteria()
        .uuid("8b7a881c-6ccb-4ada-8f6a-60cc99e6aa20")
        .actionIn("BAN", "IPBAN");

    Criteria or = uuidBan.expiresAbsent()
        .or().with(uuidBan).expiresGreaterThan(TimeInstant.of(1467364749679L));
```

see
https://github.com/immutables/immutables/blob/master/mongo/test/org/immutables/mongo/fixture/ent/Ent.java

Just to note, our criteria is basically a singly-linked flat list and `.or()` delimits parts of DNF, so that all constrains are implicitly conjunctive, with `or` acting as a separator of disjunctive parts. , therefore  `.or()`  `.with()` notation instead of combiners like `a.or(b)` `a.and(b)`. It's a peculiarity, but is unlikely to change.

**Comment by Favorlock** [comment]
Ok, thanks for your efforts in working towards making this possible. :)

**Comment by elucash** [comment]
released in 2.2.9. Thank you!
